src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -272,27 +282,41 @@ export class NetworkAreaDiagramViewer {
         if (svg != null) {
             svg.style.cursor = 'grabbing';
         }
-        this.initialPosition = DiagramUtils.getPosition(this.selectedElement); // used for the offset
-        this.edgeAngles = new Map<string, number>();
+        if (!(event as MouseEvent).shiftKey) {
+            this.initialPosition = DiagramUtils.getPosition(
```

Comment:
you could also include `this.ctm = ...` line in that if, as it's not needed for the shift selection

Reply:
line of code moved inside the if